service/worker/archiver/archiver.go

```diff
@@ -114,14 +114,38 @@ func handleRequest(ctx workflow.Context, logger log.Logger, metricsClient metric
 	}
 	actCtx := workflow.WithActivityOptions(ctx, ao)
 	uploadSW := metricsClient.StartTimer(metrics.ArchiverScope, metrics.ArchiverUploadWithRetriesLatency)
-	if err := workflow.ExecuteActivity(actCtx, uploadHistoryActivityFnName, request).Get(actCtx, nil); err != nil {
-		logger.Error("failed to upload history, moving on to deleting history without archiving", tag.Error(err))
+	err := workflow.ExecuteActivity(actCtx, uploadHistoryActivityFnName, request).Get(actCtx, nil)
+	if err != nil {
+		logger.Error("failed to upload history, will delete all uploaded blobs and moving on to deleting history without archiving", tag.Error(err))
 		metricsClient.IncCounter(metrics.ArchiverScope, metrics.ArchiverUploadFailedAllRetriesCount)
 	} else {
 		metricsClient.IncCounter(metrics.ArchiverScope, metrics.ArchiverUploadSuccessCount)
 	}
 	uploadSW.Stop()
 
+	if err != nil {
+		// If uploading or version check fails, delete all uploaded blobs
```

Comment:
This comment is a little misleading. Delete blobs should not care if its deleting blobs because upload failed or version check failed. If version check failed uploadHistory will return an error. So we just blindly delete uploaded blobs on any error from upload - which might be a version check error.

Reply:
Deleted. You are right it's misleading. I did this because when I wrote this code, I thought the version check may be done outside the upload activity.